Tests/Negotiation/FormatNegotiatorTest.php

```diff
@@ -103,6 +103,45 @@ public function testGetBestWithFormat()
         $this->assertEquals(new Accept('application/json;version=1.0'), $this->negotiator->getBest(''));
     }
 
+    public function testGetBestWithPreferExtension()
+    {
+        $priorities = ['text/html', 'application/json'];
+        $this->addRequestMatcher(true, ['priorities' => $priorities, 'prefer_extension' => '2.0']);
+
+        $reflectionClass = new \ReflectionClass(get_class($this->request));
+        $reflectionProperty = $reflectionClass->getProperty('requestUri');
+        $reflectionProperty->setAccessible(true);
+        $reflectionProperty->setValue($this->request, 'http://example.com/file.json');
+
+        // Without extension mime-type in Accept header
+
+        $this->request->headers->set('Accept', 'text/html');
+        $this->assertEquals(new Accept('application/json'), $this->negotiator->getBest(''));
+
+        // With low q extension mime-type in Accept header
+
+        $this->request->headers->set('Accept', 'text/html, application/json; q=0.1');
```

Comment:
This test caught an issue before I made my modifications. Prior to my modification of FormatNegotiator, the mime-type in the Accept header would override the one prefer_extension appends. So if you requested `file.json` with the header in that test, the final header would be:

`Accept: text/html, application/json; q=0.1, application/json; q=2.0`

It would ignore the second instance of application/json and prefer text/html, failing that test.

This test guards against the case where a contributor refactors FormatNegotiator to append the mime-type again. However if there are other guards against that, I agree we can scrap this test.